### PR TITLE
Reduce dynamic entity to basic fields

### PR DESCRIPTION
**Pull request by Prokyonn**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

Adjusts the database columns to have a consistent data set.

#### Why?

The database table is much more readable with less columns and some columns had keywords for MySql 8 as their name.

#### To Do

- [ ] Update CHANGELOG.md
- [ ] Create documentation

### CODE REVIEW

File: Entity/Dynamic.php [hunk 1]
```diff
@@ -239,36 +124,10 @@ public function getData(): array
...
-        if (!is_string($name)) {
-            return false;
-        }
-
-        if (property_exists($this, $name) && $this->$name !== null) {
-            return true;
-        }
-
-        $data = $this->getData();
+        $array = $this->getData();
```
Comment (alexander-schranz) [inline Entity/Dynamic.php:127]: move this to to a `setField(string $key, $value)` function which is called here

File: UPGRADE.md [hunk 2]
```diff
@@ -2,6 +2,271 @@
 
 ## 2.0.0 (unreleased)
 
+### Database
+
+Dynamic Entity has been reduced to some basic fields. All previous data fields are merged into the `data` column.
+
+#### Data Migration
+Migrate the data fields into the json `data`.
```
Comment (alexander-schranz) [inline UPGRADE.md:10]: add empty line before and after this line

File: UPGRADE.md [hunk 3]
```diff
@@ -2,6 +2,271 @@
...
+        ),
+        ''
+      ),
+      '}'
+    )
+  );
+
+```
+
+Delete the redundant columns.
```
Comment (alexander-schranz) [inline UPGRADE.md:241]: add empty line after this line

File: UPGRADE.md [hunk 4]
```diff
@@ -2,6 +2,271 @@
...
+  DROP COLUMN text,
+  DROP COLUMN textarea,
+  DROP COLUMN date,
+  DROP COLUMN attachment,
+  DROP COLUMN checkbox,
+  DROP COLUMN checkboxMultiple,
+  DROP COLUMN dropdown,
+  DROP COLUMN dropdownMultiple,
+  DROP COLUMN radioButtons;
+
```
Comment (alexander-schranz) [inline UPGRADE.md:267]: remove this empty

**Comment by alexander-schranz** [comment]
@Prokyonn as this is use break its also the best that we rename in this pull request the composer package from `sulu/sulu-form-bundle` to `sulu/form-bundle` https://github.com/sulu/SuluFormBundle/issues/223 (documentation need to be adjusted).

File: UPGRADE.md [hunk 5]
```diff
@@ -2,6 +2,271 @@
...
+          '\"radioButtons\":',
+          replace(dyn.radioButtons, '"', '\\\"'),
+          '[]'
+        ),
+        ''
+      ),
+      '}'
+    )
+  );
+
```
Comment (alexander-schranz) [inline UPGRADE.md:238]: remove this empty line

File: UPGRADE.md [hunk 6]
```diff
@@ -2,6 +2,271 @@
...
+      IF(
+        STRCMP('', SUBSTRING(dyn.data, 2, LENGTH(dyn.data) -2)) = 0,
+        ' ',
+        ','
+      ),
+      IF(
+        dyn.salutation is not NULL,
+        CONCAT(
+          '\"salutation\":\"',
+          replace(dyn.salutation, '"', '\\\"'),
```
Comment (alexander-schranz) [inline UPGRADE.md:28]: what happens if a text includes `\`?

Comment (Prokyonn) [reply UPGRADE.md]: Now I additionally escaped `\` and `/` are there any other symbols that should be escaped ?

File: UPGRADE.md [hunk 7]
```diff
@@ -2,6 +2,267 @@
...
+            replace(replace(replace(dyn.attachment, '"', '\\\"'), '/', '\/'), '\\', '\\\\')
+          ),
+          NULL
+        ),
+        IF(
+          dyn.checkbox is not NULL,
+          CONCAT(
+            '\"checkbox\":\"',
+            replace(replace(replace(dyn.checkbox, '"', '\\\"'), '/', '\/'), '\\', '\\\\'),
+            '\"'
```
Comment (alexander-schranz) [inline UPGRADE.md:193]: Checkbox should not be a string just 0 or 1

File: UPGRADE.md [hunk 8]
```diff
@@ -2,6 +2,266 @@
...
+            '\"attachment\":',
+            replace(replace(replace(dyn.attachment, '"', '\\\"'), '/', '\/'), '\\', '\\\\')
+          ),
+          NULL
+        ),
+        IF(
+          dyn.checkbox is not NULL,
+          CONCAT(
+            '\"checkbox\":',
+            dyn.checkbox
```
Comment (alexander-schranz) [inline UPGRADE.md:192]: @Prokyonn  Sorry it looks like checkbox can be an empty string so I think we need todo here a case when

**Pull request merged**